_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp

```diff
@@ -500,8 +500,8 @@ namespace UMC_MPEG2_DECODER
         }
 
         const auto picExt = *m_currHeaders.picExtHdr;
-        uint32_t fieldIndex = m_currFrame->GetNumberByParity(picExt.picture_structure == BOTTOM_FLD_PICTURE);
-        MPEG2DecoderFrameInfo & info = *m_currFrame->GetAU(fieldIndex);
+        m_fieldIndex = m_currFrame->GetNumberByParity(picExt.picture_structure == BOTTOM_FLD_PICTURE);
+        MPEG2DecoderFrameInfo & info = *m_currFrame->GetAU(m_fieldIndex);
```

Comment:
Not fully clear what the values if m_fieldIndex is. Can you please explain?

Reply:
when fieldIndex=0, this is a top field or frame,  `slicesInfo` will be assigned by `MPEG2DecoderFrameInfo & info = *m_currFrame->GetAU(fieldIndex)`
when m_fieldIndex = 1, this is a bottom field, `slicesInfoBottom` will be assigned by `MPEG2DecoderFrameInfo & info = *m_currFrame->GetAU(fieldIndex)`
when run firstFrameSlice = m_currFrame->`GetAU(0)`->GetSlice(0), the value of `fieldIndex` defaults to 0 in `GetAU(0)`, but we get this `slicesInfoBottom` within a frame, this pointer will return null, so I use xx to get a valid `slicesInfo` or `slicesInfoBottom`.

Reply:
Sorry, probably we're talking about different things. That is the value of class member variable instead of local variable? What is the reason to keep this "local" frame specific member in "global" MPEG2Decoder class in may contain lots of frames with a different field parity?

Reply:
> Sorry, probably we're talking about different things. That is the value of class member variable instead of local variable? What is the reason to keep this "local" frame specific member in "global" MPEG2Decoder class which may contain lots of frames with a different field parity?

because this error clip only has the top field and no bottom field, the top field information should be used in [MPEG2Decoder::IsFieldOfCurrentFrame](https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp#L846), but for the program to run, I use the bottom field information, we get the top field, bottom field and frame slice info through this local variable fieldIndex, in order to get the slice information in [MPEG2Decoder::IsFieldOfCurrentFrame](https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp#L846), I need to define it in the class.
Or according to this error clip with no bottom field, msdk can directly return error, could you give the better suggestions? Thanks.

Reply:
My point is:
1) You shouldn't define **frame** level variable into **super-class** MPEG2Decoder. If you need a new member, you can define it inside MPEG2DecoderFrame, not globally
2) You can field index locally:
```
    bool MPEG2Decoder::IsFieldOfCurrentFrame() const
    {
        auto fieldIndex = m_currFrame->GetNumberByParity(picExt.picture_structure == BOTTOM_FLD_PICTURE);
        const auto firstFrameSlice = m_currFrame->GetAU(fieldIndex )->GetSlice(0);
```
1st point is a strong request. 2nd point is optional.

Reply:
thanks, code updated.